### PR TITLE
Notification Comment details: enable feature for release

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref: #17790 

This enables the new Comment details view for Comment Notifications.

To test:

TLDR - verify everything works as expected.
- Go to Notifications > comment notification.
- Verify the new details view is displayed.
- Verify Spam, Trash, and deleted comments are removed from the notifications list.
- Verify the ⬆️ and ⬇️ nav bar buttons display the previous and next notification.
- Verify Edit, Reply, and Like work as expected.

<kbd>comment_details</kbd>

## Regression Notes
1. Potential unintended areas of impact
Should be none.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested Notification Comments and My Site Comments to be sure both work as expected.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by dvdchr — APPROVED** [review]
LGTM! :shipit:
